packages/fxa-settings/src/components/InputPhoneNumber/index.tsx

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import InputText from '../InputText';
+import { useFtlMsgResolver } from '../../models';
+import { InputModeEnum } from '../FormVerifyCode';
+import { UseFormMethods } from 'react-hook-form';
+
+interface Country {
+  id: number;
+  code: string;
+  classNameFlag: string;
+  name: string;
+  ftlId: string;
+  validationPattern: RegExp;
+}
+
+// North American Numbering Plan (NANP) countries are 123-123-1234.
+// This allows for any and all characters so users can type spaces,
+// dashes, parenthesis etc. but keeps the number of digits to 10.
+export const phoneValidationNorthAmerica = /^(?:\D*\d){10}\D*$/;
```

Comment:
Alternate regex that is a bit more specific (courtesy of ChatGPT):
```/^\(?\d{3}\)?[ .-]?\d{3}[ .-]?\d{4}$/```

Allows:
1234567890 | ✅ | Plain digits.
123-456-7890 | ✅ | Hyphen-separated.
123.456.7890 | ✅ | Dot-separated.
123 456 7890 | ✅ | Space-separated.
(123) 456-7890 | ✅ | Parentheses with spaces/hyphens.

Does not allow
(123)-456-7890	❌	Parentheses and hyphen mix.
123/456/7890	❌	Slashes are not allowed.
123-45-6789	❌	Only 9 digits.
123.456-7890	❌	Inconsistent separators.
abc-456-7890	❌	Contains letters.

Reply:
Sure! Will update, thanks; wasn't sure how specific we wanted to be here.

Reply:
I'm just wary of allowing _any_ character in the input, the above is just a suggestion. Not sure if there's other common patterns for north american phone numbers

Reply:
Oh, but do we want to block users for "(123)-456-7890 ❌ Parentheses and hyphen mix." and " 123.456-7890 ❌ Inconsistent separators."? I can see us calling it invalid but we can also just pull the numbers. 🤷‍♀️ Happy to go with your suggestion though.